src/System.Net.Sockets/src/System/Net/Sockets/SafeSocketHandle.Windows.cs

```diff
@@ -96,18 +96,33 @@ internal bool SkipCompletionPortOnSuccess
 
         internal static SafeSocketHandle CreateWSASocket(AddressFamily addressFamily, SocketType socketType, ProtocolType protocolType)
         {
-            return CreateSocket(InnerSafeCloseSocket.CreateWSASocket(addressFamily, socketType, protocolType));
+            SafeSocketHandle result = Interop.Winsock.WSASocketW(addressFamily, socketType, protocolType, IntPtr.Zero, 0, Interop.Winsock.SocketConstructorFlags.WSA_FLAG_OVERLAPPED | Interop.Winsock.SocketConstructorFlags.WSA_FLAG_NO_HANDLE_INHERIT);;
+            if (result.IsInvalid)
+            {
+                result.SetHandleAsInvalid();
+            }
+
+            if (NetEventSource.IsEnabled) NetEventSource.Info(null, result);
+            return result;
         }
```

Comment:
Does this helper need to live on SafeSocketHandle?  Seems like this code could just move into SocketPal.CreateSocket and avoid the need for the unnecessary level of indirection.  Same for the other similar accept helper here.  And on Unix.

Reply:
We need to call the protected `SetHandle`.

Reply:
So how about just adding:
```C#
internal new void SetHandle(IntPtr ptr) => base.SetHandle(ptr);
```
to the SafeSocketHandle?